config/opal_mca.m4

```diff
@@ -167,29 +167,34 @@ of type-component pairs.  For example, --enable-mca-no-build=pml-ob1])
     # Second, set the DSO_all and STATIC_all variables.  conflict
     # resolution (prefer static) is done in the big loop below
     #
+    # Exception here is the components of the accelerator framework,
+    # which by default are built to be dynamic.
+    #
     AC_MSG_CHECKING([which components should be run-time loadable])
     if test "$enable_static" != "no"; then
         DSO_all=0
         msg="none (static libraries built)"
     elif test "$OPAL_ENABLE_DLOPEN_SUPPORT" = 0; then
         DSO_all=0
         msg="none (dlopen disabled)"
-    elif test -z "$enable_mca_dso"; then
-        DSO_all=0
-        msg=default
     elif test "$enable_mca_dso" = "no"; then
         DSO_all=0
         msg=none
     elif test "$enable_mca_dso" = "yes"; then
         DSO_all=1
         msg=all
     else
+        if test -z "$enable_mca_dso"; then
+            enable_mca_dso="accelerator,btl-smcuda,rcache-gpusm,rcache-rgpusm"
+            AC_MSG_NOTICE([building accelerator framework compnents as DSOs (default)])
+        fi
         DSO_all=0
         ifs_save="$IFS"
         IFS="${IFS}$PATH_SEPARATOR,"
         msg=
         for item in $enable_mca_dso; do
-            AS_VAR_SET([AS_TR_SH([DSO_$item])], [1])
+            str="`echo DSO_$item=1 | sed s/-/_/g`"
+            eval $str
```

Comment:
Why did you remove the `AS_VAR_SET`?  Was this code broken for `foo-var` kinds of `$item` values?

Reply:
yes it was

Reply:
one of the reasons for dalcinl frustrations with building certain components (as opposed to frameworks for which the previous logic worked) as dso.s. i forgot which pr/issue that was though where he expressed that frustration with main and 5.0.x

Reply:
Can you provide an example of what this `AS_VAR_SET` + `AS_TR_SH` was handling wrong?  The point of these macros is to ensure safe handling of indirection and not-valid-for-variable-name characters.  E.g., `AS_TR_SH` handles the transmorgification of `-` to `_`.

Reply:
https://unix.stackexchange.com/questions/428880/list-of-acceptable-initial-characters-for-a-bash-variable

Reply:
you are wrong.  - remains - when run through AS_TR_SH

Reply:
well i retract that statement.  doing more testing.